src/condor_utils/stl_string_utils.h

```diff
@@ -139,22 +141,59 @@ void randomlyGenerateShortLivedPassword(std::string &str, int len);
 // unchanged during iteration.  This is trivial for string literals, of course.
 class StringTokenIterator {
 public:
-	StringTokenIterator(const char *s = NULL, int res=40, const char *delim = ", \t\r\n" ) : str(s), delims(delim), ixNext(0) { current.reserve(res); };
-	StringTokenIterator(const std::string & s, int res=40, const char *delim = ", \t\r\n" ) : str(s.c_str()), delims(delim), ixNext(0) { current.reserve(res); };
+	StringTokenIterator(const char *s = NULL, int res=40, const char *delim = ", \t\r\n" ) : str(s), delims(delim), ixNext(0), pastEnd(false) { current.reserve(res); };
+	StringTokenIterator(const std::string & s, int res=40, const char *delim = ", \t\r\n" ) : str(s.c_str()), delims(delim), ixNext(0), pastEnd(false) { current.reserve(res); };
 
-	void rewind() { ixNext = 0; }
+	void rewind() { ixNext = 0; pastEnd = false;}
 	const char * next() { const std::string * s = next_string(); return s ? s->c_str() : NULL; }
-	const char * first() { ixNext = 0; return next(); }
+	const char * first() { ixNext = 0; pastEnd = false; return next(); }
 	const char * remain() { if (!str || !str[ixNext]) return NULL; return str + ixNext; }
 
 	int next_token(int & length); // return start and length of next token or -1 if no tokens remain
 	const std::string * next_string(); // return NULL or a pointer to current token
 
+	// Allow us to use this as a bona-fide STL iterator
+	using iterator_category = std::input_iterator_tag;
+	using value_type        = std::string;
+	using difference_type   = std::ptrdiff_t;
+	using pointer           = std::string *;
+	using reference         = std::string *;
+
+	StringTokenIterator begin() const {
+		StringTokenIterator sti{str, 0, delims};
+		sti.next();
+		return sti;
+	}
+
+	StringTokenIterator end() const {
+		StringTokenIterator sti{str, 0, delims};
+		sti.ixNext = strlen(str);
+		sti.pastEnd = true;
+		return sti;
+	}
+
+	std::string &operator*() {
+		return current;
+	}
+	
+	StringTokenIterator &operator++() {
+		next();
+		return *this;
+	}
+
+	bool operator!=(const StringTokenIterator &rhs) {
+		return (this->ixNext != rhs.ixNext) || (this->pastEnd != rhs.pastEnd);
+	}
+	bool operator==(const StringTokenIterator &rhs) {
+		return this->ixNext == rhs.ixNext && this->pastEnd == rhs.pastEnd;
+	}
+
 protected:
 	const char * str;   // The string we are tokenizing. it's not a copy, caller must make sure it continues to exist.
 	const char * delims;
 	int ixNext;
 	std::string current;
+	bool pastEnd;
```

Comment:
I know size is not really an issue for these objects, but putting the pastEnd before current does make it smaller.